src/main/kotlin/com/manimdsl/frontend/SemanticAnalysis.kt

```diff
@@ -506,4 +507,25 @@ class SemanticAnalysis {
             incompatibleInitialisation("${arrayPrefix}$dataStructureType", ctx)
         }
     }
+
+    fun forLoopRangeTypeCheck(
+        symbolTable: SymbolTableVisitor,
+        startExpr: ExpressionNode,
+        endExpr: ExpressionNode,
+        ctx: ManimParser.RangeHeaderContext
+    ) {
+        val startExprType = inferType(symbolTable, startExpr)
+        val endExprType = inferType(symbolTable, endExpr)
+        if (!(startExprType is NumberType && endExprType is NumberType)
+            && !(startExprType is CharType && endExprType is CharType)) {
```

Comment:
Should this not be `||` in the middle?

Reply:
I think it should be `&&` because we should allow (both numbers `||` both characters) and to check for the error we do `!` that which results in `&&` in the middle? Correct me if I'm wrong though I think my logic might be wrong

Reply:
Oh yes, I see. This is really minor but maybe pull the `!` outside and make the middle operator a `||`? I think it would be more readable